diff --git a/modules/pt_profile/manifests/pt_tools_postboot_config.pp b/modules/pt_profile/manifests/pt_tools_postboot_config.pp
index dd0e38b..cc33c78 100644
--- a/modules/pt_profile/manifests/pt_tools_postboot_config.pp
+++ b/modules/pt_profile/manifests/pt_tools_postboot_config.pp
@@ -59,7 +59,7 @@ class pt_profile::pt_tools_postboot_config {
       $os_user            = $component_details['os_user']
       $db_settings        = $component_details['db_settings']
       $db_connect_pwd     = $db_settings['db_connect_pwd']
-      validate_hash($db_settings)
+      validate_legacy(Hash, 'validate_hash', $db_settings)  # <umaritimus dpkpatch/>
       $db_settings_array  = join_keys_to_values($db_settings, '=')
       notify {"POST-Boot Tools Component ${component_name} config Database settings\n":}
 
