diff --git a/modules/pt_profile/manifests/pt_system.pp b/modules/pt_profile/manifests/pt_system.pp
index d4c6c69..cc09053 100644
--- a/modules/pt_profile/manifests/pt_system.pp
+++ b/modules/pt_profile/manifests/pt_system.pp
@@ -21,7 +21,7 @@
 class pt_profile::pt_system {
   $is_root = inline_template('<%= Puppet.features.root?%>')
 
-  if $::osfamily != 'windows' and $is_root == "true" {
+  if $facts['os']['family'] != 'windows' and $is_root == "true" {  # <umaritimus dpkpatch/>
     $ensure = hiera('ensure')
 
     $env_type = hiera('env_type')    
