diff --git a/modules/pt_profile/manifests/pt_mssql_setup.pp b/modules/pt_profile/manifests/pt_mssql_setup.pp
index bccb3ac..b5f7692 100644
--- a/modules/pt_profile/manifests/pt_mssql_setup.pp
+++ b/modules/pt_profile/manifests/pt_mssql_setup.pp
@@ -29,7 +29,7 @@ class pt_profile::pt_mssql_setup {
   if $env_type != 'midtier' {
     fail('The pt_mssql_setup profile can only be applied to midtier env_type')
   }
-  if $::osfamily != 'windows' {
+  if $facts[os][family] != 'windows' { # <umaritimus dpkpatch/>
     fail('The pt_mssql_setup profile can only be applied on windows platforms')
   }
   $mssql_server_list = hiera('mssql_server_list')
